pkgs/development/tools/misc/gnum4/default.nix

```diff
@@ -1,19 +1,36 @@
-{ lib, stdenv, fetchurl }:
+{ lib, stdenv, fetchurl, autoreconfHook, texinfo4 }:
 
 # Note: this package is used for bootstrapping fetchurl, and thus
 # cannot use fetchpatch! All mutable patches (generated by GitHub or
 # cgit) that are needed here should be included directly in Nixpkgs as
 # files.
-
-stdenv.mkDerivation rec {
-  pname = "gnum4";
+let
   version = "1.4.19";
-
   src = fetchurl {
     url = "mirror://gnu/m4/m4-${version}.tar.bz2";
     sha256 = "sha256-swapHA/ZO8QoDPwumMt6s5gf91oYe+oyk4EfRSyJqMg=";
   };
 
+  documentation = stdenv.mkDerivation {
+    inherit version src;
+    pname = "gnum4-doc";
+
+    nativeBuildInputs = [ texinfo4 ];
+
+    buildPhase = ''
+      makeinfo --html --no-split doc/m4.texi
+    '';
```

Comment:
We can easily inline this and call it in postBuild. This would simplify this change greatly.

Reply:
I know. I just hoped that by moving building of documentation outside of main derivation, I would be able to work-around the issue. Not the case.